src/common/string/filter/sequence-matching.ts

```diff
@@ -10,10 +10,13 @@ import { fuzzyScore } from "./filter";
  * @return {number} Score representing how well the word matches the filter. Return of 0 means no match.
  */
 
-export const fuzzySequentialMatch = (filter: string, ...words: string[]) => {
+export const fuzzySequentialMatch = (
+  filter: string,
+  item: ScorableTextItem
```

Comment:
Why this changes? We dont use the item for anything else here?

Reply:
See other comment. When we do letter highlighting, we want to store the text decorations on each item as their match scores are being calculated during filtering.

Here's the relevant line of code in the PR for letter highlighting where we will use `item`:
https://github.com/home-assistant/frontend/pull/8868/files#diff-692eb3ca3265b7ffde8a1d7e15fc059f21c80d88ef1105fd046e0a0c6ad3da4cR82